src/bikeracks/templates/scripts.js

```diff
@@ -316,10 +311,18 @@ BikeMap.prototype.findAddress = function(lat, lng) {
     });
 };
 
-BikeMap.prototype.addTempMarker = function(lat, lng, userId, address) {
+BikeMap.prototype.addTempMarker = function(lat, lng, address) {
     // add a temporary marker, that is removed as soon as you click away
     //build icon
+    let userId;
     
+    if (this.auth.currentUser) {
+        userId = this.auth.currentUser.uid;
+    }
+    else {
+        userId = ""
+    }
+
```

Comment:
This is equivalent to
`let userId = this.auth.currentUser.uid || "";`.

Or if you think that's unreadable,
`let userId = this.auth.currentUser.uid ? this.auth.currentUser.uid : "";`

Or if the "false" version of this.auth.currentUser.uid is always just the empty string anyway then you can do 
`let userId = this.auth.currentUser.uid`.

Reply:
`let userId = this.auth.currentUser ? this.auth.currentUser.uid : "";`

Reply:
Done.